ballerina/engine.bal

```diff
@@ -242,54 +244,64 @@ isolated class Engine {
         }
     }
 
-    isolated function resolve(Context context, Field 'field) returns anydata {
+    isolated function resolve(Context context, Field 'field, boolean executeLoadMethod = true) returns anydata {
         parser:FieldNode fieldNode = 'field.getInternalNode();
-        parser:RootOperationType operationType = 'field.getOperationType();
+
+        if executeLoadMethod {
+            string loadMethodName = getLoadMethodName(fieldNode.getName());
+            service object {}? serviceObject = 'field.getServiceObject();
+            if serviceObject is service object {} { 
+                boolean isRemoteMethod = 'field.getOperationType() == parser:OPERATION_MUTATION && 'field.isRootField();
```

Comment:
Do we need to check whether this is a root field here? It must be a root field right?

Reply:
"The field can have a `get` accessor but can become part of a mutation operation if this field is defined in a Ballerina service object. The only time the field corresponds to a `remote` method is when it becomes a root field (i.e., when it's defined in `graphql:Service`).